WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+                    getString(R.string.no_network_message),
+                    R.drawable.ic_notice_white_64dp,
+                    getString(R.string.cancel));
+            ft.add(nuxAlert, "alert");
+            ft.commitAllowingStateLoss();
+            return false;
+        }
+        return true;
+    }
+
+    protected void next(String email) {
+        if (!checkNetworkConnectivity()) {
+            return;
+        }
+
+        if (isValidEmail(email)) {
+            showEmailCheckProgressDialog();
+            mDispatcher.dispatch(AccountActionBuilder.newIsAvailableEmailAction(email));
+        } else {
+            showEmailError(R.string.email_invalid);
+        }
+    }
+
+    private String getEmail() {
+        return EditTextUtils.getText(mEmailEditText).trim();
+    }
+
+    private boolean isValidEmail(String email) {
+        Pattern emailRegExPattern = Patterns.EMAIL_ADDRESS;
+        Matcher matcher = emailRegExPattern.matcher(email);
+
+        return matcher.find() && email.length() <= MAX_EMAIL_LENGTH;
+    }
+
+    private final OnClickListener mNextClickListener = new OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            next(getEmail());
+        }
+    };
+
+    @Override
+    public void afterTextChanged(Editable s) {
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        UpdateNextButton();
+        mEmailEditTextLayout.setError(null);
+    }
+
+    private void UpdateNextButton() {
+        if (fieldsFilled()) {
+            mNextButton.setEnabled(true);
+        } else {
+            mNextButton.setEnabled(false);
+        }
+    }
+
+    private boolean fieldsFilled() {
+        return EditTextUtils.getText(mEmailEditText).trim().length() > 0;
+    }
+
+    protected boolean isUserDataValid() {
+        final String email = EditTextUtils.getText(mEmailEditText).trim();
+        boolean retValue = true;
+
+        if (TextUtils.isEmpty(email)) {
+            showEmailError(R.string.required_field);
+            retValue = false;
+        }
+
+        return retValue;
+    }
+
+    private void showEmailError(int messageId) {
+        mEmailEditTextLayout.setError(getString(messageId));
+        mEmailEditText.post(new Runnable() {
+            @Override
+            public void run() {
+                mEmailEditText.requestFocus();
+            }
+        });
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mDispatcher.register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        mDispatcher.unregister(this);
+    }
+
+    private void showEmailCheckProgressDialog() {
+        startProgress(getActivity().getString(R.string.checking_email));
+    }
+
+    @Override
```

Comment:
I just didn't notice that we had those in `AbstractFragment`, this was my bad :(